ibis/common/grounds.py

```diff
@@ -142,13 +148,17 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({argstring})"
 
     def __eq__(self, other) -> bool:
+        # compare types
         if type(self) is not type(other):
             return NotImplemented
-
-        return all(
-            getattr(self, name, None) == getattr(other, name, None)
-            for name in self.__attributes__
-        )
+        # compare arguments
```

Comment:
These changes are also less important, only relevant for the mutable `Annotable` objects (`Config` mostly).